warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -2248,6 +2271,19 @@ else if (IvaratorRequiredVisitor.isIvaratorRequired(queryTree) && !config.canHan
         return new Tuple2<>(ranges, needsFullTable);
     }
     
+    private void attemptToRecordPlan(ShardQueryConfiguration config, JexlNode queryTree) {
+        if (null == this.plannedScript) {
```

Comment:
with this mechanism, the query tree from only the first call with actually set the query string.  Don't we want the query string updated each time?

Reply:
I considered updating the query string each time it's updated but thought it might an expense we do not want to incur.  That's why I opted for setting it only if there was an error.  I'm open to changing the approach if the reviewers agree.

Reply:
Basically what we will get here is merely a jexl translation of the original query.  What I think we want to see is the query plan just prior to the failure.  I think the approach needs to be changed here.

Reply:
RE: original query
This method only gets called if there's an error.  If the create is successful, plannedScript is initialized at the end of planning.  If an error occurs, this method will be called and it'll use the current state of queryTree.  The test cases demonstrate that this often reflects changes to the original query.
The null check probably isn't necessary here as it is almost always going to be null.

Do you want me to remove the null check?  Or would you rather plannedScript get updated after each modification to queryTree?

Reply:
@jwomeara and @hlgp agreed that it would be worthwhile to update the planned script at each step along the way, which I'm understanding to mean every time the queryTree is modified.  I'll change it to do that.